ITKTIFFModule can't configure

### Description
I'm trying to build ITK git from scratch. There is a _/usr/lib/libtiff.so.6.2.0_ ist from the _libtiff_ 4.7.1 package on my system. The configuration fails with:
```
 -- Configuring done (58.6s)
CMake Error at CMake/ITKModuleMacros.cmake:396 (target_link_libraries):
  The link interface of target "ITKTIFFModule" contains:

    ITK::optimized

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.

Call Stack (most recent call first):
  Modules/ThirdParty/TIFF/CMakeLists.txt:31 (itk_module_impl)


-- Generating done (1.5s)
```

### Steps to Reproduce

build recipe from https://gitlab.archlinux.org/archlinux/packaging/packages/itk/-/blob/main/PKGBUILD?ref_type=heads but not Vanilla Arch

```
make ..$cmake_options..
cmake --build build
```


### Actual behavior
`./build/lib/cmake/ITK-6.0/Modules/ITKTIFF.cmake` contains the erroneous

`set(ITKTIFF_LIBRARIES "ITK::optimized;/usr/lib/libtiff.so.6.2.0;ITK::debug")`

not deriving from https://github.com/InsightSoftwareConsortium/ITK/tree/main/Modules/ThirdParty/TIFF/src/itktiff

### Versions

7d80605fd737907066bf6cb3d0732ca341755ae6

### Environment
Python 3.10
CMake 3.31.6

--- a/Modules/ThirdParty/TIFF/CMakeLists.txt
+++ b/Modules/ThirdParty/TIFF/CMakeLists.txt
@@ -8,9 +8,23 @@ if(ITK_USE_SYSTEM_TIFF)
     ${ITKTIFF_BINARY_DIR}/src
     ${TIFF_INCLUDE_DIRS}
   )
-  set(ITKTIFF_SYSTEM_INCLUDE_DIRS)
-  set(ITKTIFF_LIBRARIES "${TIFF_LIBRARIES}")
+  set(ITKTIFF_LIBRARIES "TIFF::TIFF")
   set(ITKTIFF_NO_SRC 1)
+
+  set(
+    ITKTIFF_EXPORT_CODE_INSTALL
+    "
+  find_package(TIFF REQUIRED)
+  "
+  )
+  set(
+    ITKTIFF_EXPORT_CODE_BUILD
+    "
+  if(NOT ITK_BINARY_DIR)
+    find_package(TIFF REQUIRED)
+  endif()
+  "
+  )
 else()
   set(
     ITKTIFF_INCLUDE_DIRS

